derive_builder/examples/validator_fn.rs

```diff
@@ -0,0 +1,37 @@
+//! This example illustrates the use of `validator` to add a pre-build validation
+//! step.
+
+#[macro_use]
+extern crate derive_builder;
+
+#[derive(Builder, Debug, PartialEq)]
+#[builder(build_fn(validator="LoremBuilder::validate"))]
+struct Lorem {
+    #[builder(default="42")]
+    pub ipsum: u8,
+}
+
+impl LoremBuilder {
+    /// Check that `Lorem` is putting in the right amount of effort.
+    fn validate(&self) -> Result<(), String> {
+        if let Some(ref ipsum) = self.ipsum {
+            match *ipsum {
+                i if i < 20 => Err("Try harder".to_string()),
+                i if i > 100 => Err("You'll tire yourself out".to_string()),
+                _ => Ok(())
+            }
+        } else {
+            Ok(())
+        }
+    }
+}
+
+fn main() {
+    // If we don't set the field `ipsum`,
+    let x = LoremBuilder::default().build().unwrap();
```

Comment:
But this `main` does not demonstrate when or where the validator is called, so I would not say it demonstrates the functionality very explicitly. To me it looks more like a example for the `default` feature.